[sw/silicon_creator] Add spx signature field to manifest

alphan opened the pull request:
This PR renames the current signature field in the manifest to `rsa_signature` and adds a new `spx_signature` field. This new field will be revisited when we start working on manifest extensions.

Note: Marked as draft until I update rust definitions

jadephilipoom commented on sw/device/silicon_creator/lib/sigverify/spx_key.h:
```diff
@@ -24,8 +24,30 @@ enum {
    * Length of an SPX public key in words.
    */
   kSigVerifySpxKeyWords = kSigVerifySpxKeyBytes / sizeof(uint32_t),
+  /**
+   * Length of an SPX signature in bits.
+   */
+  kSigverifySpxSigNumBits = 62848,
+  /**
+   * Length of an SPX signature in bytes.
+   */
+  kSigverifySpxSigNumBytes = kSigverifySpxSigNumBits / 8,
+  /**
+   * Length of an SPX signature in words.
+   */
+  kSigverifySpxSigNumWords = kSigverifySpxSigNumBytes / sizeof(uint32_t),
```
In case it makes sense to you: I think you can probably pull the signature size from `sw/device/silicon_creator/lib/sigverify/sphincsplus/params.h` and/or `sw/device/silicon_creator/lib/sigverify/sphincsplus/verify.h`.

jadephilipoom reviewed (approved):
LGTM so far, just a few nits.

alphan commented on sw/device/silicon_creator/lib/manifest.h:
```diff
@@ -211,21 +218,22 @@ typedef struct manifest {
-OT_ASSERT_MEMBER_OFFSET(manifest_t, code_start, 884);
-OT_ASSERT_MEMBER_OFFSET(manifest_t, code_end, 888);
-OT_ASSERT_MEMBER_OFFSET(manifest_t, entry_point, 892);
+OT_ASSERT_MEMBER_OFFSET(manifest_t, spx_signature, 0);
+OT_ASSERT_MEMBER_OFFSET(manifest_t, rsa_signature, 7856);
```
It would be nice but I added it at the start on purpose: this way don't need to touch the RSA signature verification code. We will revisit this when we work on the manifest again.

cfrantz commented on sw/device/silicon_creator/lib/manifest.h:
```diff
@@ -121,7 +122,13 @@ enum {
-   * Image signature.
+   * SPX signature of the image.
+   *
+   * TODO: Make this optional.
```
Nit: there should be an issue number associated with this.

alphan replied:
Thanks for the reminder, created #17824.

alphan commented:
Thanks @jadephilipoom and @cfrantz. I addressed your comments, created an issue to revisit the manifest layout, and updated opentitantool. This PR should be ready to go once CI is done.
